packages/store/addon/-private/system/store.ts

```diff
@@ -1413,7 +1438,8 @@ const Store = Service.extend({
     // short circuit if we are already loading
     if (internalModel && internalModel.isLoading()) {
       return internalModel._promiseProxy.then(() => {
-        return internalModel.getRecord();
+        // TODO is this necessary?
+        return internalModel!.getRecord();
```

Comment:
can get rid of this non-null assertion, if `internalModel` is a const declaration instead of let